lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java

```diff
@@ -0,0 +1,523 @@
+      String details) {
+    TaskHistoryEvent historyEvent = new TaskHistoryEvent();
+    historyEvent.setId(IdGenerator.generateWithPrefix(IdGenerator.ID_PREFIX_TASK_HISTORY_EVENT));
+    historyEvent.setUserId(userid);
+    historyEvent.setDetails(details);
+    historyEvent.setWorkbasketKey(workbasketKey);
+    historyEvent.setTaskId(taskId);
+    historyEvent.setEventType(type);
+    historyEvent.setOldValue(previousWorkbasketId);
+    return historyEvent;
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class SimpleHistoryCleanupJobAllCompletedSameParentBusinessIsTrue
+      implements TaskanaConfigurationModifier {
+    @TaskanaInject TaskanaEngine taskanaEngine;
+    SimpleHistoryServiceImpl historyService;
+
+    @Override
+    public Builder modify(Builder builder) {
+      return builder.simpleHistoryCleanupJobAllCompletedSameParentBusiness(true);
+    }
+
+    @WithAccessId(user = "businessadmin")
+    @BeforeAll
+    void setup() throws Exception {
+      historyService = new SimpleHistoryServiceImpl();
+      historyService.initialize(taskanaEngine);
+    }
+
+    @Test
+    @WithAccessId(user = "admin")
+    void should_CleanHistoryEventsUntilDate_When_SameParentBusinessTrueAndEventsQualified()
+        throws Exception {
+
+      TaskHistoryEvent eventToBeCleaned =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId1",
+              TaskHistoryEventType.CREATED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned2 =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId1",
+              TaskHistoryEventType.COMPLETED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned2.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned2.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned3 =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId2",
+              TaskHistoryEventType.CREATED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned3.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned3.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned4 =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId2",
+              TaskHistoryEventType.CANCELLED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned4.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned4.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned5 =
```

Comment:
Do we need 6 events? Or would 3 be enough? It would be more readable

Reply:
I will reduce it to 4 events. 3 is not possible, because a task needs minimum 2 events to be qualified for deletion, the CREATED and e.g COMPLETED event, if not then it means the task is still active